### DISCUSSION TITLE
Integration of 2 or more Multiome Datasets

### DISCUSSION THREAD

**Discussion by sylestiel** [stuart-lab/signac#1488] [Q&A] [answered]
Hi @timoast,
I realize that this is a question often asked. However, I'm not at all clear about it reading the different posts.
I'm following the vignette, 'Joint RNA and ATAC analysis: 10x multiomic' in Signac with a view to integrating Multiome datasets. However, I'm not clear on **when** to integrate the multiome datasets. Do we have to integrate prior to the Peak Calling Step?
How do we keep the peak ranges being compared the same for the multiple datasets? Could you provide the necessary steps to do so?
Is Gene Expression data processing  and DNA Accessibility data processing done on the integrated dataset or on each of the multiome dataset first?

Thank you!

**Comment by timoast** [comment, marked as answer]
> Do we have to integrate prior to the Peak Calling Step?

For any analysis involving multiple samples, you should always start by defining one set of features (peaks) to quantify and analyze across all samples.

> How do we keep the peak ranges being compared the same for the multiple datasets? Could you provide the necessary steps to do so?

Define a set of peaks and quantify those in each of the samples. See the merging vignette for example steps: https://stuartlab.org/signac/articles/merging

> Is Gene Expression data processing and DNA Accessibility data processing done on the integrated dataset or on each of the multiome dataset first?

I would suggest doing QC on each dataset first, then integrate/merge and run joint analysis

**Comment by annepureddy** [comment]
Hi @timoast 

I am also trying to integrate multiple samples that have 10X multiome: scATAC and scRNA data. From this post, I was able to understand that I need to merge the peaks sets from each of the samples for ATAC and am following the above vignette.(https://stuartlab.org/signac/articles/merging) However, I am running into the following error: 

>frags.FC013_1 <- CreateFragmentObject(path = "~/Desktop/Multiome_Data/FC013-1/atac_fragments.tsv.gz", cells = rownames(md.FC013_1))

Computing hash
Checking for 726821 cell barcodes
**Error in CreateFragmentObject(path = "~/Desktop/Multiome_Data/FC013-1/atac_fragments.tsv.gz",  : 
  Not all cells requested could be found in the fragment file.**  

Could it be that the size of my metadata file and frag file are different? Also, in the function CreateFragmentObject do the cells variable have to be gex barcodes or ATAC barcodes? In my csv file, I have both.

It would be great if you could please help me solve this issue! 

Here is the most of my code for reference. Thank you! 

peaks.FC013_1 <- FC013_1$Peaks
peaks.FC012_2 <- FC012_2$Peaks
peaks.FC008_2 <- FC008_2$Peaks
peaks.FC011_1 <- FC011_1$Peaks
peaks.FC007 <- FC007$Peaks

convert to genomic ranges
gr.FC013_1 <- StringToGRanges(rownames(peaks.FC013_1), sep = c(":", "-"))
gr.FC012_2 <- StringToGRanges(rownames(peaks.FC012_2), sep = c(":", "-"))
gr.FC008_2 <- StringToGRanges(rownames(peaks.FC008_2), sep = c(":", "-"))
gr.FC011_1 <- StringToGRanges(rownames(peaks.FC011_1), sep = c(":", "-"))
gr.FC007 <- StringToGRanges(rownames(peaks.FC007), sep = c(":", "-"))

Create a unified set of peaks to quantify in each dataset
combined.peaks <- reduce(x = c(gr.FC007, gr.FC008_2, gr.FC011_1, gr.FC012_2, gr.FC013_1))

Filter out bad peaks based on length
peakwidths <- width(combined.peaks)
combined.peaks <- combined.peaks[peakwidths  < 10000 & peakwidths > 20]
combined.peaks

CREATE FRAGMENT FILES
load metadata
md.FC013_1 <- read.table(file = "~/Desktop/Multiome_Data/FC013-1/per_barcode_metrics.csv",stringsAsFactors = FALSE,sep = ",", header = TRUE,row.names = 1)[-1, ] 
md.FC012_2<- read.table(file = "~/Desktop/Multiome_Data/FC012-2/per_barcode_metrics.csv",stringsAsFactors = FALSE,sep = ",", header = TRUE,row.names = 1)[-1, ] 
md.FC008_2 <- read.table(file = "~/Desktop/Multiome_Data/FC008-2/per_barcode_metrics.csv",stringsAsFactors = FALSE,sep = ",", header = TRUE,row.names = 1)[-1, ] 
md.FC011_1 <- read.table(file = "~/Desktop/Multiome_Data/FC011-1/per_barcode_metrics.csv",stringsAsFactors = FALSE,sep = ",", header = TRUE,row.names = 1)[-1, ] 
md.FC007 <- read.table(file = "~/Desktop/Multiome_Data/FC007/per_barcode_metrics.csv",stringsAsFactors = FALSE,sep = ",", header = TRUE,row.names = 1)[-1, ] 

create fragment objects
frags.FC013_1 <- CreateFragmentObject(path = "~/Desktop/Multiome_Data/FC013-1/atac_fragments.tsv.gz", cells = rownames(md.FC013_1))
frags.FC012_2 <- CreateFragmentObject(path = "~/Desktop/Multiome_Data/FC012-2/atac_fragments.tsv.gz",cells = rownames(md.FC012_2))
frags.FC008_2 <- CreateFragmentObject(path = "~/Desktop/Multiome_Data/FC008-2/atac_fragments.tsv.gz",cells = rownames(md.FC008_2))
frags.FC011_1 <- CreateFragmentObject(path = "~/Desktop/Multiome_Data/FC011-1/atac_fragments.tsv.gz",cells = rownames(md.FC011_1))
frags.FC007 <- CreateFragmentObject(path = "~/Desktop/Multiome_Data/FC007/atac_fragments.tsv.gz",cells = rownames(md.FC007))

**Reply by timoast** [reply]
please search "Not all cells requested could be found in the fragment file" in the closed issues for suggestions
